### PR TITLE
[RIC] 1.4.2 List Behavior

### PR DESCRIPTION
**Pull request by bishalkg**
Added horizontal Carousel, scrolls one card at a time (need to optimize this later).
Also changed how the outfit List displays. There is a card to add the current Product, and when that product card appears, you can delete it from that list. This is all visual, the persistence of these changes by the user upon page refresh is to be implemented in a later ticket. 

Also look over the changes we all made (like in the CSS and others) that we want changed in main and make a comment and I'll change them all here before I merge.

### CODE REVIEW

File: client/dist/styles.css [hunk 1]
```diff
@@ -2,11 +2,12 @@
   box-sizing: border-box;
 }
 
+html {
+  font-size: clamp(12px, 0.6vw, 36px);
```
Comment (ya-liu) [inline client/dist/styles.css]: 24px for max?

File: client/dist/styles.css [hunk 2]
```diff
@@ -221,26 +222,100 @@ body {
...
+/* .carousel-content > * {
+  width: 100%;
+  flex-shrink: 0;
+  flex-grow: 1;
+} */
+
+
+.left-arrow, .right-arrow {
+  position:absolute;
+  z-index: 9000;
```
Comment (ya-liu) [inline client/dist/styles.css]: Do the arrows appear on the page at all times? If so, could you please dial the z-index down a little so they don't bleed through my review modals? :grimacing:

File: client/dist/styles.css (hunk 1 above)
Comment (dho1994) [reply client/dist/styles.css:6]: /*
Clamping our HTML here will define the minimum font size unit for our app when shrunken and when full-screen.
All children will have > 1 rem. That way, they'll never be smaller than the minimum bounds we're defining here.

MIN should be the minimum font size for all elements when shrunken
VALUE should be the minimum font size for all elements when full-screen
MAX doesn't matter too much right now
*/

html {
  font-size: clamp(8px, 0.6vw, 24px);
}

File: client/dist/styles.css [hunk 3]
```diff
@@ -2,11 +2,12 @@
   box-sizing: border-box;
 }
```
Comment (dho1994) [inline client/dist/styles.css:4]: /*
Clamping our HTML here will define the minimum font size unit for our app when shrunken and when full-screen.
All children will have > 1 rem. That way, they'll never be smaller than the minimum bounds we're defining here.

MIN should be the minimum font size for all elements when shrunken
VALUE should be the minimum font size for all elements when full-screen
MAX doesn't matter too much right now
*/
html {
  font-size: clamp(8px, 0.6vw, 24px);
}

File: client/dist/styles.css (hunk 1 above)
Comment (dho1994) [reply client/dist/styles.css]: let's go with the following if you're all okay with it:

Comment (dho1994) [reply client/dist/styles.css]: html {
  font-size: clamp(8px, 0.6vw, 24px);
}

Comment (dho1994) [reply client/dist/styles.css]: I added comments to my css file but it's a bit wordy so up to y'all if you want to include:

/*
Clamping our HTML here will define the minimum font size unit for our app when shrunken and when full-screen.
All children will have > 1 rem. That way, they'll never be smaller than the minimum bounds we're defining here.

MIN should be the minimum font size for all elements when shrunken
VALUE should be the minimum font size for all elements when full-screen
MAX doesn't matter too much right now
*/

File: client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const AddOutfitCard = ({ addToCurrOutfits }) => {
+  const makeLintHappy = addToCurrOutfits;
```
Comment (dho1994) [inline client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx]: XD

Comment (ya-liu) [inline client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx]: I didn't have to declare a new const with a function passed as a prop, this linter behavior is interesting

Comment (ya-liu) [reply client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx]: Try removing line 5 and change line 9 onClick to onClick={() => {addToCurrOutfits(yourParams)}}

File: client/src/components/RelatedItemsAndOutfit/UserOutfits/Carousel.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const Carousel = ({ prodsInfo, children }) => {
+  const [currIndex, setCurrIndex] = useState(0);
+  const [length, setLength] = useState(prodsInfo.length);
+  useEffect(() => {
+    setLength(prodsInfo.length);
+  }, [prodsInfo]);
```
Comment (ya-liu) [inline client/src/components/RelatedItemsAndOutfit/UserOutfits/Carousel.jsx:9]: Learned something new - call setState within useEffect

**Review by ya-liu — COMMENTED** [review]
looks a lot cleaner with more functionality!

File: client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx (hunk 4 above)
Comment (dho1994) [reply client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx]: could also be b/c proptypes is expecting syntax like "PropTypes.func.isRequired" vs. instanceOf(Function)?

**Review by dho1994 — COMMENTED** [review]
looks good to me, i think the only change needed is the html clamping and then should be good to go

File: client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx (hunk 4 above)
Comment (bishalkg) [reply client/src/components/RelatedItemsAndOutfit/UserOutfits/AddOutFitCard.jsx]: Oh no i just have this for testing purposes, I don't need the line hahaha

File: client/dist/styles.css (hunk 1 above)
Comment (bishalkg) [reply client/dist/styles.css]: done

File: client/dist/styles.css (hunk 2 above)
Comment (bishalkg) [reply client/dist/styles.css]: changed to 1

**Review by ya-liu — APPROVED** [review]

**Review by dho1994 — APPROVED** [review]
